src/NHibernate.Test/Async/Linq/LinqQuerySamples.cs

```diff
@@ -1372,20 +1372,6 @@ public async Task DLinqJoin10aLeftJoinAsync()
 			}
 		}
 
-		[Test]
-		public void ReplaceFunctionWithNullArgumentAsync()
-		{
-			var query = from e in db.Employees
-			            select e.FirstName.Replace(e.LastName, null);
-			List<string> results = null;
-			Assert.That(
-				async () =>
-				{
-					results = await (query.ToListAsync());
-				}, Throws.Nothing, "Expected REPLACE(FirstName, LastName, NULL) to be supported");
-			Assert.That(results, Is.Not.Null);
-		}
-
```

Comment:
I have no idea why this async generation change occurs...

Reply:
After some debugging I found out that the problem seems to be with Roslyn which loads incorrectly the NUnit `Assert.Than` symbols marking async counterparts parameters as nullable but not sync ones (`object?[]` vs `object[]`) . This seems to be an issue only for Roslyn version `3.0.0` as with the newer versions (e.g. `3.7.0`) it marks both as nullable. As Roslyn version `3.7.0` is not supported for `netcoreapp2.1`, I will add a loosely matching so that it will still match despite the nullable differences.

Reply:
So this is triggered by trageting 2.1 in the test project.Will it also happen with 3.1? If no, then it will be one reason more to target 3.1 in the test project instead of 2.1.

Reply:
> Will it also happen with 3.1?

Yes, the same issue is present when the test project is targeting `netcoreapp3.1`, not sure why, but only happens for Roslyn version `3.0.0.`. I released a new version `0.18.4` which fixes this issue by using a loosely matching when searching for async counterparts.

Reply:
Thanks again.